src/ray/util/util.h

```diff
@@ -211,25 +211,15 @@ inline __suppress_ubsan__("signed-integer-overflow") int64_t GenerateSeed() {
 }  // namespace internal
 }  // namespace ray
 
-/// A helper function to fill random bytes into the `data`.
-/// Warning: this is not fork-safe, we need to re-seed after that.
+// Warning: This method reads random bytes from /dev/urandom in order to
+// provide random bytes. Therefore, one should be mindful about performance
+// when calling it too many times.
 template <typename T>
 void FillRandom(T *data) {
   RAY_CHECK(data != nullptr);
-  auto randomly_seeded_mersenne_twister = []() {
-    std::mt19937 seeded_engine(ray::internal::GenerateSeed());
-    return seeded_engine;
-  };
-
-  // NOTE(pcm): The right way to do this is to have one std::mt19937 per
-  // thread (using the thread_local keyword), but that's not supported on
-  // older versions of macOS (see https://stackoverflow.com/a/29929949)
-  static std::mutex random_engine_mutex;
-  std::lock_guard<std::mutex> lock(random_engine_mutex);
-  static std::mt19937 generator = randomly_seeded_mersenne_twister();
   std::uniform_int_distribution<uint32_t> dist(0, std::numeric_limits<uint8_t>::max());
   for (size_t i = 0; i < data->size(); i++) {
-    (*data)[i] = static_cast<uint8_t>(dist(generator));
+    (*data)[i] = static_cast<uint8_t>(dist(std::random_device("/dev/urandom")));
```

Comment:
Using `std::random_device` as a RNG is heavily discouraged, since the performance of many implementations of `std::random_device` degrades sharply once the entropy pool is exhausted. You're supposed to use it to seed a PRNG such as our previous `std::mt19937` PRNG.

Can you change this so `std::mt19937` is seeded with `std::random_device` along with another source of entropy, such as a sample from a high-resolution clock?